### PR TITLE
Show duplicate mapping attribute errors in place

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose
$subject

### Related Issues
- NA

### Related PRs
- NA

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: modules/react-components/src/components/message/message.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline modules/react-components/src/components/message/message.stories.tsx:2]: Update the year

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx [hunk 2]
```diff
@@ -146,11 +151,11 @@ export const AttributeListItem: FunctionComponent<AttributeListItemPropInterface
...
     };
 
-    const handleClaimMapping = (e) => {
-        const mappingValue = e.target.value.replace(/[^\w+$:/.]/g, "");
+    const handleClaimMapping = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const mappingValue: string = e.target?.value.replace(/[^\w+$:/.]/g, "");
 
         setMappedAttribute(mappingValue);
-        updateMapping(claimURI, mappingValue);
+        updateMapping(claimURI, mappingValue,true);
```
Comment (kavinda1995) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx:158]:
```suggestion
        updateMapping(claimURI, mappingValue, true);
```

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 3]
```diff
@@ -626,15 +640,33 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
...
+     * @param isUpdatingOnInputChange - whether the updating happens on mapping attribute input change
+     * 
      */
-    const updateClaimMapping = (claimURI: string, mappedValue: string) => {
-        const claimMappingList = [ ...claimMapping ];
+    const updateClaimMapping = (claimURI: string, mappedValue: string, isUpdatingOnInputChange?: boolean) => {
+        const claimMappingList: ExtendedClaimMappingInterface[] = [ ...claimMapping ];
 
-        claimMappingList.forEach((mapping) => {
+        const alreadySeen: Record<string,boolean> = {};
```
[the remark this answers is not in the document]
Comment (pavinduLakshan) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:649]: Fixed with a6848a2

File: modules/react-components/src/components/message/message.stories.tsx (hunk 1 above)
Comment (pavinduLakshan) [reply modules/react-components/src/components/message/message.stories.tsx]: Fixed with da3ab19

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx (hunk 2 above)
Comment (pavinduLakshan) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx]: Fixed with 64390d9

**Review by kavinda1995 — DISMISSED** [review]

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx [hunk 4]
```diff
@@ -788,6 +811,13 @@ export const AttributeSelection: FunctionComponent<AttributeSelectionPropsInterf
                                                 </Table.Body>
                                             </Table>
                                         </Grid.Row>
+                                        { duplicatedMappingValues.length > 0 && (<Message 
+                                            type="error"
+                                            content={ 
+                                                t("console:develop.features.applications.notifications."
+                                                +"updateClaimConfig" + ".error.description", 
+                                                { description: "Mapped user attributes cannot be duplicated." }) }
```
Comment (kavinda1995) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx:819]: Can't we add i18n text here?

Comment (pavinduLakshan) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx]: Fixed with 45ae58a

**Review by thivi — APPROVED** [review]

**Pull request merged**